### PR TITLE
Major feature changes with pathfinding and new boss enemies

### PR DESCRIPTION
**Pull request by kk4w4i**
This pull request includes functionality changes to the pathfinding for enemy entities, as well as new bosses added to the game. Further junit tests will be sent later this sprint.

Thank you.

### CODE REVIEW

**Review by atharvagupta2003 — APPROVED** [review]

File: saves/savefile.json
Comment (Foref) [inline saves/savefile.json:1]: json file incorrect indent

Comment (kk4w4i) [reply saves/savefile.json:1]: this is copied straight from main
its a new file addition

File: source/saves/savefile.json
Comment (Foref) [inline source/saves/savefile.json:1]: Indentation issue

File: saves/savefile.json
Comment (Foref) [reply saves/savefile.json:1]: ok, resolve merge conflicts and it will probably be fine

**Review by rowan-gray — APPROVED** [review]
Other than the player losing health on earth every second, seems fine 👍🏼

**Review by AnshulDadhwal — APPROVED** [review]
Seems fine. Just need to fix the savefile conflict...should be easy

**Pull request merged**